pkg/cli/environment.go

```diff
@@ -124,7 +134,15 @@ func (s *EnvSettings) Namespace() string {
 //RESTClientGetter gets the kubeconfig from EnvSettings
 func (s *EnvSettings) RESTClientGetter() genericclioptions.RESTClientGetter {
 	s.configOnce.Do(func() {
-		s.config = kube.GetConfig(s.KubeConfig, s.KubeContext, s.namespace)
+		clientConfig := kube.GetConfig(s.KubeConfig, s.KubeContext, s.namespace)
+		if len(s.KubeToken) > 0 {
+			clientConfig.BearerToken = &s.KubeToken
+		}
+		if len(s.KubeAPIServer) > 0 {
+			clientConfig.APIServer = &s.KubeAPIServer
+		}
```

Comment:
@bacongobbler I have updated this to have empty string checks. This should avoid any empty string security flaws the API might have.

Reply:
> This is looking great!
> 
> I want to make sure that passing an empty string (`&""` instead of `nil`) for `clientConfig.BearerToken` and `clientConfig.ApiServer` won't accidentally pass along some extra headers, tricking certain Kubernetes api server implementations.
> 
> Marking this for 3.2 for review

@bacongobbler This should solve the empty string issue.